integration_test/Utils.ts

```diff
@@ -0,0 +1,440 @@
+   * @param ostComposer OSTComposer contract address.
+   * @returns EIP712 compatible stakerequest hash.
+   */
+  public getStakeRequestHash(stakeRequest: StakeRequest, gateway: string, ostComposer: string): Buffer | Uint8Array {
+    const stakeRequestMethod = 'StakeRequest(uint256 amount,address beneficiary,uint256 gasPrice,uint256 gasLimit,uint256 nonce,address staker,address gateway)';
+    const encodedTypeHash = web3Utils.sha3(
+      this.originWeb3.eth.abi.encodeParameter('string', stakeRequestMethod),
+    );
+    const stakeIntentTypeHash = web3Utils.soliditySha3(
+      { type: 'bytes32', value: encodedTypeHash },
+      { type: 'uint256', value: stakeRequest.amount!.toString(10) },
+      { type: 'address', value: stakeRequest.beneficiary! },
+      { type: 'uint256', value: stakeRequest.gasPrice!.toString(10) },
+      { type: 'uint256', value: stakeRequest.gasLimit!.toString(10) },
+      { type: 'uint256', value: stakeRequest.nonce!.toString(10) },
+      { type: 'address', value: stakeRequest.staker! },
+      { type: 'address', value: gateway },
+    );
+
+    const EIP712_DOMAIN_TYPEHASH = web3Utils.soliditySha3(
+      'EIP712Domain(address verifyingContract)',
+    );
+    const DOMAIN_SEPARATOR = web3Utils.soliditySha3(
+      this.originWeb3.eth.abi.encodeParameters(
+        ['bytes32', 'address'],
+        [EIP712_DOMAIN_TYPEHASH, ostComposer],
+      ),
+    );
+
+    const eip712TypeData = EthUtils.keccak(
+      Buffer.concat(
+        [
+          Buffer.from('19', 'hex'),
+          Buffer.from('01', 'hex'),
+          EthUtils.toBuffer(DOMAIN_SEPARATOR),
+          EthUtils.toBuffer(stakeIntentTypeHash),
+        ],
+      ),
+    );
+
+    return EthUtils.bufferToHex(eip712TypeData);
+  }
+
+  /**
+   * It provides repository object pointing to mosaic facilitator db file.
+   * @returns Repositories object.
+   */
+  private async getRepositories(): Promise<Repositories> {
+    return Repositories.create(
+      path.join(
+        Directory.getDBFilePath('1000'),
+        'mosaic_facilitator.db',
+      ),
+    );
+  }
+
+  /**
+   * It verifies flow of stake and minting of OSTPrime.
+   * @param stakeRequestHash Hash of stake request.
+   * @returns `true` if minting of OSTPrime is done.
+   */
+  public async verifyMinting(stakeRequestHash: string): Promise<boolean> {
+    const repos: Repositories = await this.getRepositories();
+
+    const stakeRequest: StakeRequest | null = await repos.stakeRequestRepository.get(
+      stakeRequestHash,
+    );
+
+    if (stakeRequest !== null) {
+      while (!this.messageHash) {
+        await new Promise(done => setTimeout(done, 2000));
+        const sr = await repos.stakeRequestRepository.get(stakeRequest.stakeRequestHash);
+        this.messageHash = sr !== null ? sr.messageHash : undefined;
+      }
+
+      const message = await repos.messageRepository.get(this.messageHash);
+
+      switch (parseInt(this.step, 10)) {
```

Comment:
```suggestion
            switch (this.step) {
        case Step.Initial:
          if (
            message!.sourceStatus === MessageStatus.Declared
            && message!.targetStatus === MessageStatus.Undeclared
          ) {
            this.step = Step.Intermediate;
          }
          break;

        case Step.Intermediate:
          if (
            message!.sourceStatus === MessageStatus.Declared
            && message!.targetStatus === MessageStatus.Declared
          ) {
            this.step = Step.Final;
          }
          break;

        case Step.Final:
          if (
            message!.sourceStatus === MessageStatus.Progressed
            && message!.targetStatus === MessageStatus.Progressed
          ) {
            this.mintingStatus = true;
            return true;
          }
          break;
      }
```

Reply:
Why are the steps needed?
I am not able to understand it here. Cant we just return the status
`message!.sourceStatus === MessageStatus.Progressed
            && message!.targetStatus === MessageStatus.Progressed`